arches/app/media/js/viewmodels/workflow-step.js

```diff
@@ -209,16 +199,62 @@ define([
 
         this.toggleInformationBox = function() {
             var informationBoxData = self.informationBoxData();
-            var isHidden = informationBoxData['hidden'];
+            var isDisplayed = informationBoxData['displayed'];
 
-            informationBoxData['hidden'] = !isHidden;
+            informationBoxData['displayed'] = !isDisplayed;
             self.informationBoxData(informationBoxData);
 
-            self.setToLocalStorage('informationBoxHidden', !isHidden);
+            self.setMetadataToLocalStorage('informationBoxDisplayed', !isDisplayed);
         };
 
-        this.getInformationBoxHiddenStateFromLocalStorage = function() {
-            return self.getFromLocalStorage('informationBoxHidden')
+        this.getInformationBoxDisplayedStateFromLocalStorage = function() {
+            return self.getMetadataFromLocalStorage('informationBoxDisplayed')
+        };
+
+        this.getMetadataFromLocalStorage = function(key) {
+            var workflowsMetadataLocalStorageData = JSON.parse(localStorage.getItem('workflow-metadata')) || {};
+            var workflowName = ko.unwrap(config.workflow.workflowName);
+            var stepName = ko.unwrap(config.name);
+            if (workflowsMetadataLocalStorageData[workflowName] && workflowsMetadataLocalStorageData[workflowName][stepName]) {
+                return workflowsMetadataLocalStorageData[workflowName][stepName][key];
+            }
+        };
+
+        config.workflow.workflowName.subscribe(function(workflowName){
+            self.setupInformationBox(workflowName);
+        })
+
+        this.setupInformationBox = function(workflowName) {
+            if (config.informationboxdata && workflowName) {
+                var isDisplayed = true;
+                if (self.getInformationBoxDisplayedStateFromLocalStorage() !== undefined){
+                    isDisplayed = self.getInformationBoxDisplayedStateFromLocalStorage();
+                }
+                self.informationBoxData({
+                    displayed: isDisplayed,
+                    heading: config.informationboxdata['heading'],
+                    text: config.informationboxdata['text'],
+                })
+            }
+        }
+        this.setMetadataToLocalStorage = function(key, value) {
```

Comment:
This is fantastic code overall.

My one nit is that I think it's living in the wrong place.

Instead of having a listener in `workflow-step` for `workflow.workflowName`,  and then having a function to set/get localStorage that knows data about the workflow ( small red flag ), why not have this logic live in the workflow?

If it lives in the workflow we can remove the observable subscription in workflow-step. Workflow can then pass the hidden/displayed status to the workflow-step on creation as an observable, which the `workflow-step` could then manipulate. The workflow has a subscription on this observable to set to localStorage.

That keeps the `workflow-step` from knowing anything about `workflow` , and removes the need for a workflowName subscription. It still uses the good persistence logic you've written it just removes some maybe extra stuff.

Also this is just my opinion. If you feel strongly I'm happy to discuss alternatives~!